2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala

```diff
@@ -81,22 +84,97 @@ class CollectorService(
           optionalHeaderValueByName("User-Agent") { userAgent =>
             optionalHeaderValueByName("Referer") { refererURI =>
               headerValueByName("Raw-Request-URI") { rawRequest =>
-                hostName { host =>
-                  clientIP { ip =>
-                    requestInstance{ request =>
-                      entity(as[String]) { body =>
+                headerValueByName("Kinesis-Part-Key") { kinesisPartKey =>
```

Comment:
It won't collide, but maybe I can combine them into one. Originally, I had two so that I could have a separate one for use with the mobile endpoint, but now, I can probably get away with combining them. The optionalHeaderByValue only affects whether or not the param is treated as an Option or not. I don't ever check the value directly, I only do a null check so the Option distinction is of little importance although it's still probably good practice to use an option to be safe.